src/engine/enginebuffer.cpp

```diff
@@ -523,6 +533,7 @@ void EngineBuffer::slotTrackLoaded(TrackPointer pTrack,
     m_dSlipRate = 0;
     // Reset the pitch value for the new track.
     m_pause.unlock();
+    m_pTrackLoaded->setAndConfirm(1);
```

Comment:
If it's after pause.unlock() then the engine could view these COs in an inconsistent state -- best to stick it above the unlock.